core/src/main/java/com/datastrato/gravitino/storage/relational/utils/MetadataObjectUtils.java

```diff
@@ -0,0 +1,141 @@
+import com.google.common.base.Splitter;
+import java.util.List;
+import javax.annotation.Nullable;
+
+/**
+ * MetadataObjectUtils is used for converting full name to entity id and converting entity id to
+ * full name.
+ */
+public class MetadataObjectUtils {
+  private static final Splitter DOT_SPLITTER = Splitter.on('.');
+
+  private MetadataObjectUtils() {}
+
+  public static long getSecurableObjectEntityId(
+      long metalakeId, String fullName, MetadataObject.Type type) {
+    if (fullName.equals(Entity.SECURABLE_ENTITY_RESERVED_NAME)
+        && type == MetadataObject.Type.METALAKE) {
+      return Entity.ALL_METALAKES_ENTITY_ID;
+    }
+
+    if (type == MetadataObject.Type.METALAKE) {
+      return MetalakeMetaService.getInstance().getMetalakeIdByName(fullName);
+    }
+
+    List<String> names = DOT_SPLITTER.splitToList(fullName);
+    long catalogId =
+        CatalogMetaService.getInstance().getCatalogIdByMetalakeIdAndName(metalakeId, names.get(0));
+    if (type == MetadataObject.Type.CATALOG) {
+      return catalogId;
+    }
+
+    long schemaId =
+        SchemaMetaService.getInstance().getSchemaIdByCatalogIdAndName(catalogId, names.get(1));
+    if (type == MetadataObject.Type.SCHEMA) {
+      return schemaId;
+    }
+
+    if (type == MetadataObject.Type.FILESET) {
+      return FilesetMetaService.getInstance().getFilesetIdBySchemaIdAndName(schemaId, names.get(2));
+    } else if (type == MetadataObject.Type.TOPIC) {
+      return TopicMetaService.getInstance().getTopicIdBySchemaIdAndName(schemaId, names.get(2));
+    } else if (type == MetadataObject.Type.TABLE) {
+      return TableMetaService.getInstance().getTableIdBySchemaIdAndName(schemaId, names.get(2));
+    }
+
+    throw new IllegalArgumentException(String.format("Doesn't support the type %s", type));
+  }
+
+  // Securable object may be  null because the securable object may be deleted.
+  @Nullable
+  public static String getSecurableObjectFullName(String type, long entityId) {
+    if (type.equals(Entity.ALL_METALAKES_ENTITY_TYPE)) {
+      return Entity.SECURABLE_ENTITY_RESERVED_NAME;
+    }
+
+    MetadataObject.Type metadatatype = MetadataObject.Type.valueOf(type);
+    if (metadatatype == MetadataObject.Type.METALAKE) {
+      MetalakePO metalakePO = MetalakeMetaService.getInstance().getMetalakePOById(entityId);
+      if (metalakePO == null) {
+        return null;
+      }
+
+      return metalakePO.getMetalakeName();
+    }
+
+    if (metadatatype == MetadataObject.Type.CATALOG) {
+      return getCatalogFullName(entityId);
+    }
+
+    if (metadatatype == MetadataObject.Type.SCHEMA) {
+      return getSchemaFullName(entityId);
+    }
+
+    if (metadatatype == MetadataObject.Type.TABLE) {
+      TablePO tablePO = TableMetaService.getInstance().getTablePOById(entityId);
+      if (tablePO == null) {
+        return null;
+      }
+      return getSchemaFullName(tablePO.getSchemaId()) + "." + tablePO.getTableName();
```

Comment:
use DOT joiner, instead of hard code "." here.

Reply:
Done.